src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -199,25 +200,33 @@ internal class RecommendationService(
   suspend fun updateRecommendationWithManagerRecallDecision(
     jsonRequest: JsonNode?,
     recommendationId: Long,
-    readableUserName: String?
+    readableUserName: String?,
+    userLoginName: String?
   ): RecommendationResponse {
     validateManagerRecallDecision(jsonRequest)
     val existingRecommendationEntity = getRecommendationEntityById(recommendationId)
     val userAccessResponse = existingRecommendationEntity.data.crn?.let { userAccessValidator.checkUserAccess(it) }
     if (userAccessValidator.isUserExcludedRestrictedOrNotFound(userAccessResponse)) {
       throw UserAccessException(Gson().toJson(userAccessResponse))
     } else {
-      val readerForUpdating: ObjectReader = CustomMapper.readerForUpdating(existingRecommendationEntity.data)
-      val updateRecommendationRequest: RecommendationModel = readerForUpdating.readValue(jsonRequest)
-
-      existingRecommendationEntity.data.managerRecallDecision = updateRecommendationRequest.managerRecallDecision
+      val updatedRecommendation: RecommendationModel = recommendationFromRequest(existingRecommendationEntity, jsonRequest)
+      existingRecommendationEntity.data.managerRecallDecision = updatedRecommendation.managerRecallDecision
         ?.copy(
           createdDate = utcNowDateTimeString(),
           createdBy = readableUserName
         )
 
       val savedRecommendation = recommendationRepository.save(existingRecommendationEntity)
       log.info("recommendation for ${savedRecommendation.data.crn} updated with manager recall decision")
+      if (savedRecommendation.data.managerRecallDecision?.isSentToDelius == true) {
+        log.info("About to send domain event for ${savedRecommendation.data.crn} on manager recall decision made")
```

Comment:
Might be worth adding the recommendation ID to all the log messages in this method, to make it easier to find relevant log messages when all we know is the ID?

Reply:
done